src/python/pants/jvm/target_types.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    Sources,
+    StringSequenceField,
+    Target,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class JavaSources(Sources):
+    expected_file_extensions = (".java",)
+
+
+class JavaTestsSources(JavaSources):
+    default = ("*Test.java",)
+
+
+class JunitTests(Target):
+    alias = "junit_tests"
```

Comment:
Thanks for updating. Do you anticipate we will use JUnit for Scala, or stick to Scalatest? If we do use Junit, then this name would no longer work as well.

I don't like this nor think we should do it, but a truly reflective name would be `java_junit_tests`.

Reply:
I have no clue.  My best inclination is that we'll probably need to support both, eventually, so we should err on the side of whatever causes the least pain in that scenario.

Reply:
Okay, I'm not sure what the right solution is. But this is experimental for a reason.

Oh also, we want these target types defined in `pants/backend/java/target_types.py`, not `pants/jvm`. Fine to do as an unrelated change in a followup PR imo.